src/Data/Product/Properties.agda

```diff
@@ -35,4 +34,8 @@ module _ {a b} {A : Set a} {B : Set b} where
 
   ≡-dec : Decidable {A = A} _≡_ → Decidable {A = B} _≡_ →
           Decidable {A = A × B} _≡_
-  ≡-dec = PwND.≡?×≡?⇒≡?
+  ≡-dec dec₁ dec₂ (a , b) (c , d) with dec₁ a c
+  ... | no a≢c = no (a≢c ∘ ,-injectiveˡ)
+  ... | yes refl with dec₂ b d
+  ...   | no b≢d  = no (b≢d ∘ ,-injectiveʳ)
+  ...   | yes refl = yes refl
```

Comment:
Looking at the code, it just hit me: is this an instance of [`_×-dec_`](https://agda.github.io/agda-stdlib/Relation.Nullary.Product.html#585) with a bit of [map′](https://agda.github.io/agda-stdlib/Relation.Nullary.Decidable.html#2002)
to reorganise the proof. In other words:


```agda
  ,-injective : ∀ {a c : A} {b d : B} → (a , b) ≡ (c , d) → a ≡ c × b ≡ d
  ,-injective refl = refl , refl

  ≡-dec : Decidable {A = A} _≡_ → Decidable {A = B} _≡_ →
          Decidable {A = A × B} _≡_
  ≡-dec a? b? (a , b) (c , d) =
    map′ (uncurry (cong₂ _,_)) ,-injective (a? a c ×-dec b? b d)
```